gulpfile.js

```diff
@@ -285,23 +306,33 @@ gulp.task('prefetch', function(cb) {
         if(!alreadyDownloaded) {
           // download only what can be included in offline installer
           if (reqs[currentKey].bundle === 'yes') {
-            counter++;
-            console.log('DOWNLOADING -> ' + currentUrl);
-            request(currentUrl)
-              .pipe(fs.createWriteStream(currentFile)).on('finish',function() {
-                // create a sha256sum file
-                counter--;
-                if(counter===0) {
-                  cb();
-                }
-              });
+            promises.push(new Promise(function(resolve, reject) {
+              console.log('[INFO] Download ' + currentUrl + ' to ' + currentFile);
+              request(currentUrl)
+                .pipe(fs.createWriteStream(currentFile)).on('finish',function(err, res) {
+                  if (err) { 
+                    reject(err);
+                  } else {
+                    // console.log('DOWNLOADED <- ' + currentFile);
+                    promises.push(copyFromPrefetchToDist(currentFile, destFile));

```

Comment:
I think that this is causing trouble... you can't push copy promise to _promises_ array because it will be called in parallel with download task. You need to refactor it out and in the end do something like `return Promise.all(downloadFiles).then(copyFiles).then(() => cb()).catch(cb);`